crates/polars-arrow/src/io/ipc/write/writer.rs

```diff
@@ -116,7 +120,11 @@ impl<W: Write> FileWriter<W> {
         // write the schema, set the written bytes to the schema
 
         let encoded_message = EncodedData {
-            ipc_message: schema_to_bytes(&self.schema, &self.ipc_fields),
+            ipc_message: schema_to_bytes(
+                &self.schema,
+                &self.ipc_fields,
+                self.custom_schema_metadata.as_deref(),
```

Comment:
We shouldn't need to write custom metadata here - i think it causes us to write it twice since we are already writing it below in `finish()`?

Reply:
You're right, only writing the metadata to the file footer works as well. Good catch :+1: done in 78654fc09